ci/scripts/run_ci.sh

```diff
@@ -43,29 +43,46 @@ else
   exit 1
 fi
 
-pr_list_file="open_pr_list"
+pr_list_dbfile="${GFS_CI_ROOT}/open_pr_list.db"
 
-if [[ -s "${GFS_CI_ROOT}/${pr_list_file}" ]]; then
-  pr_list=$(cat "${GFS_CI_ROOT}/${pr_list_file}")
-else
-  echo "no PRs to process .. exit"
+#NOTE: last pipe with head -2 limits no more tha two PRs at a time
+pr_list=""
+if [[ -f "${pr_list_dbfile}" ]]; then
+  pr_list=$("${HOMEgfs}/ci/scripts/pr_list_database.py" --display "${pr_list_dbfile}" | grep -v Failed | grep Open | grep Built | awk '{print $1}' | head -2) || true
+fi
+if [[ -z "${pr_list}" ]]; then
+  echo "no PRs open and ready for checkout/build .. exiting"
   exit 0
 fi
 
 #############################################################
 # Loop throu all PRs in PR List and look for expirments in
 # the RUNTESTS dir and for each one run runcotorun on them
+# only upto two cases will advance at a time
 #############################################################
 
 for pr in ${pr_list}; do
   echo "Processing Pull Request #${pr} and looking for cases"
   pr_dir="${GFS_CI_ROOT}/PR/${pr}"
+  # If the directory RUNTESTS is not present then
+  # setupexpt.py has no been run yet for this PR
+  if [[ ! -d "${pr_dir}/RUNTESTS" ]]; then
+     continue
+  fi
+  num_cases=0
   for cases in "${pr_dir}/RUNTESTS/"*; do
+    if [[ ! -d "${cases}" ]]; then
+       continue
+    fi
+    ((num_cases=num_cases+1))
+    # No more than two cases are going forward at a time for each PR
+    if [[ "${num_cases}" -gt 2 ]]; then
```

Comment:
```suggestion
    if (( num_cases > max_concurrent )); then
```

Reply:
If we are already limiting `pr_list` to two cases on L51, isn't keeping track of the number of cases here unnecessary?

Reply:
OK, I'll fix this as part of making in configurable.

Reply:
one is for cases the other is for PRs, I documented that when I implemented this update and renamed the vars **max_concurrent_pr** and **max_concurrent_cases** respectively